app/src/main/java/com/cornellappdev/uplift/ui/screens/ProfileCreationScreen.kt

```diff
@@ -0,0 +1,227 @@
+                fontSize = 24.sp,
+                fontFamily = montserratFamily,
+                fontWeight = FontWeight.Bold
+            )
+            Spacer(modifier = Modifier.height(25.dp))
+
+
+
+            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    Checkbox(
+                        checked = checkedState.value,
+                        onCheckedChange = { checkedState.value = it },
+                        colors = checkboxColors
+                    )
+                    Text(
+                        text = "I agree with EULA terms and agreements",
+                        fontSize = 12.sp,
+                        fontFamily = montserratFamily,
+                        fontWeight = FontWeight.Normal
+                    )
+                }
+
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    Checkbox(
+                        checked = checkedState2.value,
+                        onCheckedChange = { checkedState2.value = it },
+                        colors = checkboxColors
+                    )
+                    Text(
+                        text = "I allow Uplift to access data on my gym usage",
+                        fontSize = 12.sp,
+                        fontFamily = montserratFamily,
+                        fontWeight = FontWeight.Normal
+                    )
+                }
+
+                Row(verticalAlignment = Alignment.CenterVertically) {
+                    Checkbox(
+                        checked = checkedState3.value,
+                        onCheckedChange = { checkedState3.value = it },
+                        colors = checkboxColors
+                    )
+                    Text(
+                        text = "I allow Uplift to access my location",
+                        fontSize = 12.sp,
+                        fontFamily = montserratFamily,
+                        fontWeight = FontWeight.Normal
+                    )
+                }
+
+            }
+
+            if (checkedState.value && checkedState2.value && checkedState3.value) {
+                Spacer(modifier = Modifier.height(155.dp))
+
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier.padding(bottom = 25.dp)
+                ) {
+                    Text(
+                        text = "Are you ready to",
+                        fontSize = 16.sp,
+                        fontFamily = montserratFamily,
+                        fontWeight = FontWeight.Bold
+                    )
+
+                    Spacer(modifier = Modifier.width(6.dp))
+
+                    Image(
+                        painter = painterResource(R.drawable.ic_main_logo),
+                        contentDescription = "",
+                        modifier = Modifier.size(width = 26.dp, height = 23.dp)
```

Comment:
I think it would be a really nice effect to make this fade in with opacity with an `animateFloatAsState` instead of kind of abruptly appearing. You can then apply `Modifier.alpha(animatedOpacity.value)` or similar to cause it to fade in instead of conditionally render.

Refer to animation lecture if ur unsure how to do this!

Reply:
Also, let's then pull this out into another `private` Composable with a `modifier: Modifier = Modifier` parameter added and then applied to the root `Row`.

```kotlin
@Composable
private fun ReadyToUplift(
    modifier: Modifier = Modifier,
) {
    Row(modifier = modifier /* Not Modifier; this is the modifier argument. Feel free to then apply other modifiers to this modifier. */) {
        // ...
    }
}
```